server/src/internalClusterTest/java/org/elasticsearch/rest/action/admin/cluster/RestNodesStatsActionIT.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.rest.action.admin.cluster;
+
+import org.elasticsearch.action.admin.cluster.allocation.TransportGetAllocationStatsAction;
+import org.elasticsearch.action.admin.cluster.node.stats.NodesStatsRequest;
+import org.elasticsearch.action.admin.cluster.node.stats.NodesStatsRequestParameters;
+import org.elasticsearch.common.util.CollectionUtils;
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.test.ESIntegTestCase;
+import org.elasticsearch.test.transport.MockTransportService;
+
+import java.util.Collection;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class RestNodesStatsActionIT extends ESIntegTestCase {
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return CollectionUtils.appendToCopy(super.nodePlugins(), MockTransportService.TestPlugin.class);
+    }
+
+    public void testDoesNotSendUnnecessaryElectedMasterNodeStatsRequest() {
```

Comment:
Does Not Send Unnecessary = Send Necessary :)
`does not` and `unnecessary` is double negation.

Reply:
That's not quite true, there's an implicit quantification here that makes these two things different:

- "does not send unnecessary" means "every stats request it sends is necessary (but it may send zero such requests)", i.e. $\neg \exists r \in \mathrm{requests}. \neg \mathrm{necessary}(r)$, or equivalently $\forall r \in \mathrm{requests}. \mathrm{necessary}(r)$.

- In contrast "sends a necessary request" means "it sends at least one necessary stats request (but it may also send one or more unnecessary ones)", i.e. $\exists r \in \mathrm{requests}. \mathrm{necessary}(r)$.

Reply:
"SendOnlyNecessaryRequests" should have the same meaning as original but with no negations

Reply:
That's even better, it means both :grin: